pkg/securelaunch/tpm/tpm.go

```diff
@@ -78,14 +138,15 @@ func ExtendPCR(tpmHandle io.ReadWriteCloser, pcr int, hash []byte) error {
 }
 
 /*
- * ExtendPCRDebug extends a PCR with the contents of a byte slice.
+ * ExtendPCRDebug extends a PCR with the contents of a byte slice
+ * and notifies the kernel of this measurement by sending event via sysfs.
  *
  * In debug mode, it prints
  * 1. old pcr value before the hash is written to pcr
  * 2. new pcr values after hash is written to pcr
  * 3. compares old and new pcr values and prints error if they are not
  */
-func ExtendPCRDebug(tpmHandle io.ReadWriteCloser, pcr int, data io.Reader) error {
+func ExtendPCRDebug(tpmHandle io.ReadWriteCloser, pcr int, data io.Reader, eventDesc []byte) error {
```

Comment:
why is eventDesc a []byte? It seems every one of the descriptions is a string and it gets cast to []byte everywhere, so you should pass in a string? The function can internally convert it to whatever it needs.

Reply:
eventDesc was added because we are trying to write u-root measurements to internal kernel buffer via sysfs and eventDesc describes what that measurement is. This internal buffer is a binary blob that has these events(a.k.a measurements stored)...Then tpmtool unmarshals this binary blob into human readable format and we store this human readable file onto disk at the end of sluinit before kexec'ing so that this file can be used for remote attestation...

Now eventDesc needs to match the marshalling format tpmtool has so that tpmtool can unmarshal it..
https://github.com/u-root/u-root/blob/60526a63fd44679cf18013d2bd6ad5fb1efb5527/vendor/github.com/9elements/tpmtool/pkg/tpm/structures.go#L125

Since event is described as a []byte in line 130, I had to make it a []byte slice. The only reason I am converting it to string is for by debug printfs (i.e. slaunch.debug calls)